p2p/conn/connection.go

```diff
@@ -452,8 +484,10 @@ FOR_LOOP:
 			c.Logger.Debug("Receive Ping")
 			c.pong <- struct{}{}
```

Comment:
This pong channel is unbuffered. I think it means we could block here if the sendRoutine exits. This might be reason why StopPeerForError doesn't work sometimes? Might also play into a leak, Would you mind thinking through that and let me know if you agree or if I'm wrong for some reason ?

Reply:
How it prevents StopPeerForError from working? if the code is inside a goroutine. Leak? yes